src/platform/packages/shared/response-ops/rule_form/src/rule_type_modal/components/index.tsx

```diff
@@ -36,6 +40,16 @@ export const RuleTypeModalComponent: React.FC<RuleTypeModalComponentProps> = ({
 }) => {
   const [selectedProducer, setSelectedProducer] = useState<string | null>(null);
   const [searchString, setSearchString] = useState<string>('');
+  const [selectedMode, setSelectedMode] = useState<'ruleType' | 'template'>('ruleType');
+
+  // Debounce search string for template API calls to avoid excessive requests
+  const [debouncedSearchString, setDebouncedSearchString] = useState<string>('');
+
+  const { run: updateDebouncedSearch } = useDebounceFn(setDebouncedSearchString, DEBOUNCE_OPTIONS);
+
+  useEffect(() => {
```

Comment:
I think we can remove this `useEffect` by setting `onChangeSearch={updateDebouncedSearch}` as a prop in the `RuleTypeModal`.